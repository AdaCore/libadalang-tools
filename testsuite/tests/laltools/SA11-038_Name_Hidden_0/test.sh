#!/bin/sh
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Bar --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Qaax --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Garply --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Bor --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Wibble --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Wobble --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Fleb --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Flib --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.ads -L 1 -R 9 -N Flob --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 12 -R 17 -N Qaax --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 12 -R 17 -N Garply --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 12 -R 17 -N Wibble --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 12 -R 17 -N Wobble --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 12 -R 17 -N Fleb --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 12 -R 17 -N Flib --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 12 -R 17 -N Flob --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 25 -R 10 -N Flib --algorithm analyse_ast
safe_rename  -P default.gpr -S ./src/foo.adb -L 25 -R 10 -N Flob --algorithm analyse_ast
