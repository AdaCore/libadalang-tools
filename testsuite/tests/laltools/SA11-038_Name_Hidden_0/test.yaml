description:
    Refactoring - Rename test that checks if renaming a definition will
    hide it by another one.
driver: shell_script
