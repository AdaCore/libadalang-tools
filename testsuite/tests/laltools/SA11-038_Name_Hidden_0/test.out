# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> bar.adb
foo.ads -> bar.ads
Problems:
Renaming Foo to Bar will hide it by <Id "Bar" foo.ads:2:4-2:7>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> qaax.adb
foo.ads -> qaax.ads
Problems:
Renaming Foo to Qaax will hide it by <Id "Qaax" foo.ads:9:7-9:11>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> garply.adb
foo.ads -> garply.ads
Problems:
Renaming Foo to Garply will hide it by <Id "Garply" foo.ads:16:10-16:16>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> bor.adb
foo.ads -> bor.ads
Problems:
Renaming Foo to Bor will hide it by <Id "Bor" foo.ads:28:4-28:7>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> wibble.adb
foo.ads -> wibble.ads
Problems:
Renaming Foo to Wibble will hide it by <Id "Wibble" foo.adb:13:7-13:13>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> wobble.adb
foo.ads -> wobble.ads
Problems:
Renaming Foo to Wobble will hide it by <Id "Wobble" foo.adb:16:10-16:16>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> fleb.adb
foo.ads -> fleb.ads
Problems:
Renaming Foo to Fleb will hide it by <Id "Fleb" foo.adb:25:10-25:14>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> flib.adb
foo.ads -> flib.ads
Problems:
Renaming Foo to Flib will hide it by <Id "Flib" foo.adb:28:13-28:17>

# Renaming node foo.ads:1:9: 
References:
Unit:foo.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 14
End_Line: 1
End_Column: 17
Line: 42
Source Location Range:
Start_Line: 42
Start_Column: 5
End_Line: 42
End_Column: 8
Unit:foo.ads
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 9
End_Line: 1
End_Column: 12
Line: 29
Source Location Range:
Start_Line: 29
Start_Column: 5
End_Line: 29
End_Column: 8
Unit:main.adb
Line: 1
Source Location Range:
Start_Line: 1
Start_Column: 6
End_Line: 1
End_Column: 9
Source Location Range:
Start_Line: 1
Start_Column: 15
End_Line: 1
End_Column: 18
File renames:
foo.adb -> flob.adb
foo.ads -> flob.ads
Problems:
Renaming Foo to Flob will hide it by <Id "Flob" foo.adb:31:16-31:20>

# Renaming node foo.adb:12:17: 
References:
Unit:foo.adb
Line: 12
Source Location Range:
Start_Line: 12
Start_Column: 17
End_Line: 12
End_Column: 20
Line: 41
Source Location Range:
Start_Line: 41
Start_Column: 8
End_Line: 41
End_Column: 11
Unit:foo.ads
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 12
End_Line: 8
End_Column: 15
Line: 24
Source Location Range:
Start_Line: 24
Start_Column: 8
End_Line: 24
End_Column: 11
Unit:main.adb
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 4
End_Line: 6
End_Column: 7
Line: 7
Source Location Range:
Start_Line: 7
Start_Column: 4
End_Line: 7
End_Column: 7
Problems:
Renaming Qux to Qaax will hide it by <Id "Qaax" foo.ads:9:7-9:11>

# Renaming node foo.adb:12:17: 
References:
Unit:foo.adb
Line: 12
Source Location Range:
Start_Line: 12
Start_Column: 17
End_Line: 12
End_Column: 20
Line: 41
Source Location Range:
Start_Line: 41
Start_Column: 8
End_Line: 41
End_Column: 11
Unit:foo.ads
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 12
End_Line: 8
End_Column: 15
Line: 24
Source Location Range:
Start_Line: 24
Start_Column: 8
End_Line: 24
End_Column: 11
Unit:main.adb
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 4
End_Line: 6
End_Column: 7
Line: 7
Source Location Range:
Start_Line: 7
Start_Column: 4
End_Line: 7
End_Column: 7
Problems:
Renaming Qux to Garply will hide it by <Id "Garply" foo.ads:16:10-16:16>

# Renaming node foo.adb:12:17: 
References:
Unit:foo.adb
Line: 12
Source Location Range:
Start_Line: 12
Start_Column: 17
End_Line: 12
End_Column: 20
Line: 41
Source Location Range:
Start_Line: 41
Start_Column: 8
End_Line: 41
End_Column: 11
Unit:foo.ads
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 12
End_Line: 8
End_Column: 15
Line: 24
Source Location Range:
Start_Line: 24
Start_Column: 8
End_Line: 24
End_Column: 11
Unit:main.adb
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 4
End_Line: 6
End_Column: 7
Line: 7
Source Location Range:
Start_Line: 7
Start_Column: 4
End_Line: 7
End_Column: 7
Problems:
Renaming Qux to Wibble will hide it by <Id "Wibble" foo.adb:13:7-13:13>

# Renaming node foo.adb:12:17: 
References:
Unit:foo.adb
Line: 12
Source Location Range:
Start_Line: 12
Start_Column: 17
End_Line: 12
End_Column: 20
Line: 41
Source Location Range:
Start_Line: 41
Start_Column: 8
End_Line: 41
End_Column: 11
Unit:foo.ads
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 12
End_Line: 8
End_Column: 15
Line: 24
Source Location Range:
Start_Line: 24
Start_Column: 8
End_Line: 24
End_Column: 11
Unit:main.adb
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 4
End_Line: 6
End_Column: 7
Line: 7
Source Location Range:
Start_Line: 7
Start_Column: 4
End_Line: 7
End_Column: 7
Problems:
Renaming Qux to Wobble will hide it by <Id "Wobble" foo.adb:16:10-16:16>

# Renaming node foo.adb:12:17: 
References:
Unit:foo.adb
Line: 12
Source Location Range:
Start_Line: 12
Start_Column: 17
End_Line: 12
End_Column: 20
Line: 41
Source Location Range:
Start_Line: 41
Start_Column: 8
End_Line: 41
End_Column: 11
Unit:foo.ads
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 12
End_Line: 8
End_Column: 15
Line: 24
Source Location Range:
Start_Line: 24
Start_Column: 8
End_Line: 24
End_Column: 11
Unit:main.adb
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 4
End_Line: 6
End_Column: 7
Line: 7
Source Location Range:
Start_Line: 7
Start_Column: 4
End_Line: 7
End_Column: 7
Problems:
Renaming Qux to Fleb will hide it by <Id "Fleb" foo.adb:25:10-25:14>

# Renaming node foo.adb:12:17: 
References:
Unit:foo.adb
Line: 12
Source Location Range:
Start_Line: 12
Start_Column: 17
End_Line: 12
End_Column: 20
Line: 41
Source Location Range:
Start_Line: 41
Start_Column: 8
End_Line: 41
End_Column: 11
Unit:foo.ads
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 12
End_Line: 8
End_Column: 15
Line: 24
Source Location Range:
Start_Line: 24
Start_Column: 8
End_Line: 24
End_Column: 11
Unit:main.adb
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 4
End_Line: 6
End_Column: 7
Line: 7
Source Location Range:
Start_Line: 7
Start_Column: 4
End_Line: 7
End_Column: 7
Problems:
Renaming Qux to Flib will hide it by <Id "Flib" foo.adb:28:13-28:17>

# Renaming node foo.adb:12:17: 
References:
Unit:foo.adb
Line: 12
Source Location Range:
Start_Line: 12
Start_Column: 17
End_Line: 12
End_Column: 20
Line: 41
Source Location Range:
Start_Line: 41
Start_Column: 8
End_Line: 41
End_Column: 11
Unit:foo.ads
Line: 8
Source Location Range:
Start_Line: 8
Start_Column: 12
End_Line: 8
End_Column: 15
Line: 24
Source Location Range:
Start_Line: 24
Start_Column: 8
End_Line: 24
End_Column: 11
Unit:main.adb
Line: 6
Source Location Range:
Start_Line: 6
Start_Column: 4
End_Line: 6
End_Column: 7
Line: 7
Source Location Range:
Start_Line: 7
Start_Column: 4
End_Line: 7
End_Column: 7
Problems:
Renaming Qux to Flob will hide it by <Id "Flob" foo.adb:31:16-31:20>

# Renaming node foo.adb:25:10: 
References:
Unit:foo.adb
Line: 25
Source Location Range:
Start_Line: 25
Start_Column: 10
End_Line: 25
End_Column: 14
Problems:
Renaming Fleb to Flib will hide it by <Id "Flib" foo.adb:28:13-28:17>

# Renaming node foo.adb:25:10: 
References:
Unit:foo.adb
Line: 25
Source Location Range:
Start_Line: 25
Start_Column: 10
End_Line: 25
End_Column: 14
Problems:
Renaming Fleb to Flob will hide it by <Id "Flob" foo.adb:31:16-31:20>

